include/spblas/vendor/rocsparse/multiply.hpp

```diff
@@ -0,0 +1,120 @@
+#pragma once
+
+#include <type_traits>
+
+#include <hip/hip_runtime.h>
+#include <rocsparse/rocsparse.h>
+
+#include <spblas/allocator.hpp>
+#include <spblas/detail/ranges.hpp>
+#include <spblas/detail/view_inspectors.hpp>
+
+#include "allocator.hpp"
+#include "types.hpp"
+
+namespace spblas {
+
+class spmv_state_t {
+public:
+  spmv_state_t() : spmv_state_t(std::make_shared<detail::rocm_allocator>()) {
+    rocsparse_create_handle(&handle_);
```

Comment:
The handle can be used for everything, we shouldn't create a handle for every state.